prime-router/src/main/kotlin/fhirengine/translation/hl7/utils/FhirPathUtils.kt

```diff
@@ -99,8 +102,14 @@ object FhirPathUtils : Logging {
             val expressionNode = parsePath(expression)
             val value = if (expressionNode == null) emptyList()
             else pathEngine.evaluate(appContext, focusResource, bundle, bundle, expressionNode)
-            if (value.size == 1 && value[0].isBooleanPrimitive) (value[0] as BooleanType).value
-            else {
+            if (value.size == 1 && value[0].isBooleanPrimitive) {
+                (value[0] as BooleanType).value
+            } else if (focusResource.isEmpty) {
```

Comment:
I can back this change out if folks don't agree.  I added this since even though I do think that the conditions should always check for existence there are a **lot** of places where they don't, but there is no issue from a test perspective because the resource is always present.

This serves as a catch all to make sure that those places don't become a problem in the future.

Reply:
Yes, I think that all sounds right, but we can actually further clean it up by checking if the evaluated value is empty or not.  I'll push that change